crates/ruff_python_formatter/src/statement/suite.rs

```diff
@@ -143,7 +143,11 @@ impl FormatRule<Suite, PyFormatContext<'_>> for FormatSuite {
         };
 
         while let Some(following) = iter.next() {
-            if is_class_or_function_definition(preceding)
+            // Add empty lines before and after a function or class definition. If the preceding
+            // node is a function or class, and contains trailing comments, then the statement
+            // itself will add the requisite empty lines when formatting its comments.
+            if (is_class_or_function_definition(preceding)
+                && !comments.has_trailing_own_line(preceding))
```

Comment:
Should the class always add the new lines rather than making it conditional on the trailing comments. It would allow removing this check and allows us to have the logic in a single place.

Reply:
Would that not require us to know if the class is the last statement in the suite, to avoid adding unnecessary trailing newlines?

Reply:
I see. Could be worth explaining this as part of the comment.